File: SDKs/motaword.com/1.0/typescript/src/sdk/models/operations/packageprojecttranslationmemoryforlanguagestatus.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PackageProjectTranslationMemoryForLanguageStatusPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=languageCode" })
  languageCode: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=projectId" })
  projectId: number;
}


export class PackageProjectTranslationMemoryForLanguageStatusQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=async_request_key" })
  asyncRequestKey: string;
}


export class PackageProjectTranslationMemoryForLanguageStatusRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PackageProjectTranslationMemoryForLanguageStatusPathParams;

  @SpeakeasyMetadata()
  queryParams: PackageProjectTranslationMemoryForLanguageStatusQueryParams;
}


export class PackageProjectTranslationMemoryForLanguageStatusResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  asyncOperationStatus?: shared.AsyncOperationStatus;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/motaword.com/1.0/python/src/sdk/models/operations/packageprojecttranslationmemoryforlanguagestatus.py
import dataclasses
from typing import Optional
from ..shared import asyncoperationstatus as shared_asyncoperationstatus


@dataclasses.dataclass
class PackageProjectTranslationMemoryForLanguageStatusPathParams:
    language_code: str = dataclasses.field(metadata={'path_param': { 'field_name': 'languageCode', 'style': 'simple', 'explode': False }})
    project_id: int = dataclasses.field(metadata={'path_param': { 'field_name': 'projectId', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class PackageProjectTranslationMemoryForLanguageStatusQueryParams:
    async_request_key: str = dataclasses.field(metadata={'query_param': { 'field_name': 'async_request_key', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class PackageProjectTranslationMemoryForLanguageStatusRequest:
    path_params: PackageProjectTranslationMemoryForLanguageStatusPathParams = dataclasses.field()
    query_params: PackageProjectTranslationMemoryForLanguageStatusQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class PackageProjectTranslationMemoryForLanguageStatusResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    async_operation_status: Optional[shared_asyncoperationstatus.AsyncOperationStatus] = dataclasses.field(default=None)
    
